generators/chipyard/src/main/scala/example/TutorialTile.scala

```diff
@@ -0,0 +1,199 @@
+    new MyTile(this, crossing, lookup)
+  }
+}
+
+// DOC include start: Tile class
+class MyTile(
+  val myParams: MyTileParams,
+  crossing: ClockCrossingType,
+  lookup: LookupByHartIdImpl,
+  q: Parameters)
+  extends BaseTile(myParams, crossing, lookup, q)
+  with SinksExternalInterrupts
+  with SourcesExternalNotifications
+{
+
+  // Private constructor ensures altered LazyModule.p is used implicitly
+  def this(params: MyTileParams, crossing: TileCrossingParamsLike, lookup: LookupByHartIdImpl)(implicit p: Parameters) =
+    this(params, crossing.crossingType, lookup, p)
+
+  // Require TileLink nodes
+  val intOutwardNode = IntIdentityNode()
+  val masterNode = visibilityNode
+  val slaveNode = TLIdentityNode()
+
+  // Implementation class (See below)
+  override lazy val module = new MyTileModuleImp(this)
+
+  // Required entry of CPU device in the device tree for interrupt purpose
+  val cpuDevice: SimpleDevice = new SimpleDevice("cpu", Seq("my-organization,my-cpu", "riscv")) {
+    override def parent = Some(ResourceAnchors.cpus)
+    override def describe(resources: ResourceBindings): Description = {
+      val Description(name, mapping) = super.describe(resources)
+      Description(name, mapping ++
+                        cpuProperties ++
+                        nextLevelCacheProperty ++
+                        tileProperties)
+    }
+  }
+
+  ResourceBinding {
+    Resource(cpuDevice, "reg").bind(ResourceAddress(hartId))
+  }
+
+  // (Connection to bus, interrupt, etc.)
+// }
+  // DOC include end: Tile class
+  
+  // DOC include start: AXI4 node
+  // # of bits used in TileLink ID for master node. 4 bits can support 16 master nodes, but you can have a longer ID if you need more. 
+  val idBits = 4
+  val memAXI4Node = AXI4MasterNode(
+    Seq(AXI4MasterPortParameters(
+      masters = Seq(AXI4MasterParameters(
+        name = "myPortName",
+        id = IdRange(0, 1 << idBits))))))
+  val memoryTap = TLIdentityNode() // Every bus connection should have their own tap node
+  // DOC include end: AXI4 node
+
+  // DOC include start: AXI4 convert
+  (tlMasterXbar.node  // tlMasterXbar is the bus crossbar to be used when this core / tile is acting as a master; otherwise, use tlSlaveXBar
+    := memoryTap
+    := TLBuffer()
+    := TLFIFOFixer(TLFIFOFixer.all) // fix FIFO ordering
+    := TLWidthWidget(masterPortBeatBytes) // reduce size of TL
+    := AXI4ToTL() // convert to TL
+    := AXI4UserYanker(Some(2)) // remove user field on AXI interface. need but in reality user intf. not needed
+    := AXI4Fragmenter() // deal with multi-beat xacts
+    := memAXI4Node) // The custom node, see below
+  // DOC include end: AXI4 convert
+
+}
+
+// DOC include start: Implementation class
+class MyTileModuleImp(outer: MyTile) extends BaseTileModuleImp(outer){
+  // annotate the parameters
+  Annotated.params(this, outer.myParams)
+
+  // TODO: Create the top module of the core and connect it with the ports in "outer" }
```

Comment:
Can you add an example of an instantiated Verilog blackbox here?

Reply:
Is that really necessary? There's already examples of Verilog blackboxes. I would point to the relevant sections in the documentation for an example of a Verilog blackbox.

Reply:
Just the instantiation, not the full wrapper

Reply:
That may make the users think they have to have a Verilog black box, which is not necessary if they have a Chisel core.

Reply:
I agree with the sentiment. However, I believe that 90% of people who will read this docs page will be trying to integrate a Verilog core, and since they are less familiar with chisel and rocket, and this that should be the audience this page should target, and it should make it easy for them to understand.
I think add that line, with an extra comment saying "this can be replaced with a complete chisel implementation of a core" should be fine

Reply:
I see Alon's point. But I would prefer not to check in another empty verilog blackbox. 
Can you add text as comment saying that people integrating Verilog cores should look at ArianeTileModuleImp, and people integrating Chisel cores should look at RocketTileModuleImp or BoomTileModuleImp.

Reply:
Fixed. I added the blackbox line as part of the comments and also described that people can use Chisel core there.